inference-engine/src/low_precision_transformations/src/assign_and_read_value.cpp

```diff
@@ -0,0 +1,142 @@
+        if (!readValue) {
+            return false;
+        }
+
+        // TODO: remove this limitation and change the transformation when this constant will be accepted to be non-zero
+        if (!NetworkHelper::isZeroConst(readValue->get_input_node_shared_ptr(0))) {
+            return false;
+        }
+
+        if (transformation_callback(op)) {
+            return false;
+        }
+        return transform(*context, m);
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(
+        std::make_shared<pattern::op::Or>(OutputVector{ assign3, assign6 }),
+        "AssignAndReadValueTransformation");
+    this->register_matcher(m, callback);
+}
+
+bool AssignAndReadValueTransformation::transform(TransformationContext& context, ngraph::pattern::Matcher& m) {
+    if (!canBeTransformed(context, m.get_match_root())) {
+        return false;
+    }
+
+    const auto oldAssign = m.get_match_root();
+    const auto readValue = oldAssign->get_control_dependencies()[0];
+    oldAssign->remove_control_dependency(readValue);
+
+    const auto assign = NetworkHelper::separateInStandaloneBranch(oldAssign);
+    const auto dequantization = NetworkHelper::getDequantization(assign);
+
+    auto oldVar = ov::as_type_ptr<op::ReadValueBase>(readValue)->get_variable();
+    auto variable_info = oldVar->get_info();
+    oldVar->update({variable_info.data_shape, dequantization.data.get_element_type(), variable_info.variable_id});
+    // transform ReadValue part
+    const auto newConstant = foldConvert(readValue->get_input_node_shared_ptr(0), dequantization.data.get_element_type());
+    const auto newReadValue = readValue->copy_with_new_inputs({newConstant});
+    const auto newDequantization = dequantization.copyWithNewInput(newReadValue);
+    replace_node(readValue, newDequantization);
+
+    // transform Assign part
+
+    const auto newAssign = assign->copy_with_new_inputs({dequantization.data});
+    function->remove_sink(as_type_ptr<op::Sink>(oldAssign));
+    function->add_sinks({as_type_ptr<op::Sink>(newAssign)});
+
+    NetworkHelper::copyInfo(assign, newAssign);
+    replace_node(assign, newAssign);
+    newAssign->add_control_dependency(newReadValue);
+
+    // fuse dequantization multiply with FQ after ReadValue if possible
+    const auto nextLayers = newDequantization->get_output_target_inputs(0);
+    if (nextLayers.size() > 1) {
+        return true;
+    }
+    const auto fakeQuantize = as_type_ptr<opset1::FakeQuantize>(nextLayers.begin()->get_node()->shared_from_this());
+
+    if (fakeQuantize == nullptr) {
+        return true;
+    }
+    auto fakeQuantizeInputs = fakeQuantize->input_values();
+
+    const auto inputLow = as_type_ptr<opset1::Constant>(fakeQuantizeInputs[1].get_node_shared_ptr());
+    const auto inputHigh = as_type_ptr<opset1::Constant>(fakeQuantizeInputs[2].get_node_shared_ptr());
+
+    if (inputLow == nullptr || inputHigh == nullptr) {
```

Comment:
There really should be || ? I thought about &&

Reply:
thanks for comment, it's not an issue: both operations have to be `Constant`. If at least one operation is not `Constant` then we can not handle `FakeQuantize` operation later.